daemons/controld/controld_utils.c

```diff
@@ -706,16 +706,20 @@ update_dc(xmlNode * msg)
         if (AM_I_DC && !pcmk__str_eq(welcome_from, fsa_our_uname, pcmk__str_casei)) {
             invalid = TRUE;
 
-        } else if (fsa_our_dc && !pcmk__str_eq(welcome_from, fsa_our_dc, pcmk__str_casei)) {
+        } else if ((controld_globals.dc_name != NULL)
+                   && !pcmk__str_eq(welcome_from, controld_globals.dc_name,
+                                    pcmk__str_none)) {
             invalid = TRUE;
         }
 
         if (invalid) {
-            CRM_CHECK(fsa_our_dc != NULL, crm_err("We have no DC"));
+            CRM_CHECK(controld_globals.dc_name != NULL,
+                      crm_err("We have no DC"));
```

Comment:
I think this should just be the first if condition rather than an assertion

Reply:
I think it should be removed altogether. Since `invalid` is true and `dc_name` is NULL, we know that:
1. We are the/a DC, and
2. The `welcome_from` node isn't us.

At that point, we might as well just log the `if (AM_I_DC)` message and forget about logging the "We have no DC" message. (Maybe it signals a problem(?) but it isn't really relevant to the flow here.

Reply:
Works for me